api_tests/lib/actors/actor.ts

```diff
@@ -849,18 +873,98 @@ export class Actor {
         }
     }
 
-    public async createLnInvoice(sats: number) {
-        return this.wallets.lightning.createInvoice(sats);
+    /// This is to be removed once cnd supports lightning
+    public lnCreateSha256Secret(): { secret: string; secretHash: string } {
+        const secretBuf = Buffer.alloc(32);
+        for (let i = 0; i < secretBuf.length; i++) {
+            secretBuf[i] = Math.floor(Math.random() * 255);
+        }
+
+        const secretHash = sha256(secretBuf);
+        const secret = secretBuf.toString("hex");
+        this.logger.debug(`LN: secret: ${secret}, secretHash: ${secretHash}`);
+        return { secret, secretHash };
     }
 
-    public async payLnInvoice(request: string) {
-        return this.wallets.lightning.pay(request);
+    public async lnCreateHoldInvoice(
+        sats: string,
+        secretHash: string,
+        expiry: number
+    ): Promise<void> {
+        this.logger.debug("LN: Create Hold Invoice", sats, secretHash, expiry);
+        const resp = await this.wallets.lightning.inner.addHoldInvoice(
+            sats,
+            secretHash,
+            expiry,
+            "test hold invoice"
+        );
+        this.logger.debug("LN: Create Hold Response:", resp);
     }
 
-    public async assertLnInvoiceSettled(id: string) {
-        const resp = await this.wallets.lightning.getInvoice(id);
-        if (!resp.is_confirmed) {
-            throw new Error(`Invoice ${id} is not confirmed}`);
+    public async lnSendPayment(
+        to: Actor,
+        satAmount: string,
+        secretHash: string,
+        finalCltvDelta: number
+    ) {
+        const toPubkey = await to.wallets.lightning.inner.getPubkey();
+        this.logger.debug(
```

Comment:
Have you verified that these log statements output what you think they output? I am not so sure here :D

Reply:
Oh yeah absolutely, it has been key in debugging some hiccups:
`[2020-03-04T10:37:13.228] [DEBUG] alice - LN: Send Payment - to: 03e8be6f8ccc7574402aa750de5fd0737de3aeca5dbaf474798449a72e038e298e ; amt: 10000 ; hash: e35d2ed167ea0b9dae31c88afa270b5714d73920e58e30cc645cb3002848ea4f ; finalCltvDelta:  30`